marketplace/service.go

```diff
@@ -140,6 +140,35 @@ func (service *Service) GetNFTDataByCardID(ctx context.Context, cardID uuid.UUID
 	return lotData, ErrMarketplace.Wrap(err)
 }
 
+// GetApproveByCardID returns nft data by card id from DB.
+func (service *Service) GetApproveByCardID(ctx context.Context, cardID uuid.UUID) (nfts.TokenIDWithApproveData, error) {
```

Comment:
maybe you need one method , and in method you can check cardID and get or not get tokenID